source/particle/property/interface.cc

```diff
@@ -220,6 +227,23 @@ namespace aspect
       }
 
       template <int dim>
+      std::vector<UpdateFlags>
+      Manager<dim>::get_needed_update_flags () const
+      {
+        const unsigned int n_components = this->introspection().n_components;
+        std::vector<UpdateFlags> update (n_components,update_default);
+        for (typename std::list<std_cxx11::shared_ptr<Interface<dim> > >::const_iterator
+             p = property_list.begin(); p!=property_list.end(); ++p)
+          {
+            const std::vector<UpdateFlags> property_update ((*p)->get_needed_update_flags());
+
+            for (unsigned int i = 0; i<n_components; ++i)
+              update[i] |= property_update[i];
```

Comment:
I see now that I was mistaken because the flags are per-field component, not per-property. maybe make this clear with a comment if you want to keep that scheme.